apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.Diagnostics.Common;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    /// Middleware that will, when invoked, call the next <see cref="RequestDelegate"/>,
+    /// and trace the time taken for the next delegate to run.  The time taken and metadata
+    /// will be sent to the Stackdriver Trace API.
+    /// </summary>
+    public sealed class CloudTraceMiddleware
```

Comment:
Is the simple name ever logged without the full namespace? I'm wondering whether `GoogleCloudTraceMiddleware` might be clearer if so. Probably unnecessary, but just something to consider.